lessons/module-3/react-class-components.md

```diff
@@ -0,0 +1,360 @@
+          onChange={event => this.handleChange(event)}
+        />
+        <input
+          type='text'
+          placeholder='Description'
+          name='description'
+          value={this.state.description}
+          onChange={event => this.handleChange(event)}
+        />
+        <button>SUBMIT</button>
+      </form>
+    )
+}
+
+```
+
+<section class="note">
+
+### For your consideration: 
+
+It's common to define event handlers with arrow functions. Why do you think that is?
+
+</section>
+
+<section class="call-to-action">
+
+### Your turn!
+
+Use your new class component knowledge to begin refactoring the Form in the Ideabox repository into a functional component. Work with a partner to do the following:
+
+- import the `useState` hook from the React library
+- refactor the class based component into a functional component of the same name
+- remove the old constructor, and replace it with calls to the useState hook (make sure to name both your new piece of state, and the function that will update it)
+- update your render method into a return statement to make use of your new state and setter methods
+- NOTE: The current form makes use of a handleChange method. You don’t necessarily need this, but you can use it with some appropriate refactoring.
+
+</section>
+
+<section class="answer">
+
+### Here's one way you could do it, don't look until you're done!
+
+```jsx
+import { useState } from 'react';
+import './Form.css';
+
+function Form(props) {
+  const [title, setTitle] = useState('')
+  const [description, setDescription] = useState('')
+
+  const submitIdea = event => {
+    event.preventDefault();
+    const newIdea = {
+      id: Date.now(),
+      title,
+      description
+    }
+    props.addIdea(newIdea);
+    clearInputs();
+  }
+
+  const clearInputs = () => {
+    setTitle('')
+    setDescription('')
+  }
+
+  return (
+    <form>
+      <input
+        type='text'
+        placeholder='Title'
+        name='title'
+        value={title}
+        onChange={event => setTitle(event.target.value)}
+      />
+
+      <input
+        type='text'
+        placeholder='Description'
+        name='description'
+        value={description}
+        onChange={event => setDescription(event.target.value)}
+      />
+
+      <button onClick={submitIdea}>SUBMIT</button>
+    </form>
+  )
+}
+
+export default Form;
+
+```
+
+</section>
+
+## Lifecycle Methods
+
+In functional components, you can call functions in response to the lifecycle of a component using the `useEffect` hook. 
+
+Review the following questions before diving into Lifecycle methods for class components: 
+
+- How do you call a function when the component mounts using `useEffect`?
+- How do you call a function when state changes using `useEffect`? 
+
+Class components inherit several lifecycle methods that we can use. Two important methods are: 
+
+`componentDidMount` - this method is invoked immediately after a component is mounted
```

Comment:
Maybe include something here about how React invokes this for us. We just write the declaration, defining what we want to happen at the moment when React invokes it.

Reply:
Great catch! I think that is something that is extra confusing, so this addition will be helpful.